onadata/apps/main/forms.py

```diff
@@ -324,6 +331,13 @@ def publish(self, user, id_string=None, created_by=None):
             else:
                 project = self._project
 
+            if 'xml_file' in self.cleaned_data and\
```

Comment:
same as above, if `xml_file in self.cleaned_data` is true, isn't `self.cleaned_data['xml_file']` also true? can't we remove the second part